### PR TITLE
counting only non empty libraries when the viewer is not self and adding...

### PR DESCRIPTION
**Pull request by eishay**
... some tests

### CODE REVIEW

**Comment by eishay** [comment]
@2is10

File: kifi-backend/modules/shoebox/app/com/keepit/model/UserConnectionRepo.scala [hunk 1]
```diff
@@ -211,7 +218,10 @@ class UserConnectionRepoImpl @Inject() (
           friendRequestRepo.save(friendRequest.copy(state = FriendRequestStates.IGNORED))
         }
 
-      (users + userId) foreach invalidateCache
+      users foreach { friend =>
+        deleteCacheForUsers(userId, friend)
+      }
```
Comment (2is10) [inline kifi-backend/modules/shoebox/app/com/keepit/model/UserConnectionRepo.scala:223]: This pattern will repeat 5 calls specifically for just `userId` for each friend (those in `invalidateCache`). We could optimize that repetition away if it’s worth it.

File: kifi-backend/modules/shoebox/test/com/keepit/controllers/website/UserProfileControllerTest.scala [hunk 2]
```diff
@@ -322,6 +339,7 @@ class UserProfileControllerTest extends Specification with ShoeboxTestInjector {
         status(result3) must equalTo(OK)
         contentType(result3) must beSome("application/json")
         val resultJson3 = contentAsJson(result3)
+        println(s"resultJson3=$resultJson3")
```
Comment (2is10) [inline kifi-backend/modules/shoebox/test/com/keepit/controllers/website/UserProfileControllerTest.scala:342]: remove?

**Comment by 2is10** [comment]
:+1: Great find, thanks for fixing.

File: kifi-backend/modules/shoebox/app/com/keepit/model/UserConnectionRepo.scala (hunk 1 above)
Comment (eishay) [inline kifi-backend/modules/shoebox/app/com/keepit/model/UserConnectionRepo.scala:223]: right

File: kifi-backend/modules/shoebox/test/com/keepit/controllers/website/UserProfileControllerTest.scala (hunk 2 above)
Comment (eishay) [inline kifi-backend/modules/shoebox/test/com/keepit/controllers/website/UserProfileControllerTest.scala:342]: sure

**Pull request merged**